x-pack/plugins/remote_clusters/plugin.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { API_BASE_PATH } from './common';
+import { CoreSetup } from './shim';
+import {
+  registerGetRoute,
+  registerAddRoute,
+  registerUpdateRoute,
+  registerDeleteRoute,
+} from './server/routes/api';
+
+export class Plugin {
+  public setup(core: CoreSetup): void {
+    const {
+      http: { createRouter, isEsError },
+    } = core;
+
+    const router = createRouter(API_BASE_PATH);
+
+    // Register routes.
+    registerGetRoute(router);
+    registerAddRoute(router);
+    registerUpdateRoute(router);
+    registerDeleteRoute(router, isEsError);
```

Comment:
I don't love that this method signature is different than the other ones. I would prefer one `registerRoutes()` method call that calls all the other ones [like in SR](https://github.com/elastic/kibana/blob/master/x-pack/plugins/snapshot_restore/server/routes/api/register_routes.ts). any additional helpers can then be passed as part of `Plugins` interface, or maybe a `Core` interface in this case.